### PR TITLE
Clarify Kubernetes Getting Started guide

### LINKED ISSUE #9086: Misc. clarification items in the "Getting Started – Kubernetes with SSO" guide
## Details
Based on working through the ["Getting Started – Kubernetes with SSO"](https://goteleport.com/docs/kubernetes-access/getting-started/cluster/) guide, here are some ways we could make it a bit clearer:

- From the "Getting started" page (https://goteleport.com/docs/getting-started/), the "Kubernetes" link took me to the "Cluster" page (https://goteleport.com/docs/kubernetes-access/getting-started/cluster/) in the "Kubernetes Access" sidebar section. However, there is also an "Agent" page in that section. I could guess that the "Cluster" setup was preferred, but at this point, I didn’t know this for sure.

- The first "Tip" box: **Edit:** Since these are related to a reusable partial, let's save these comments for another issue.

  - ~The terms "token UUIDs" and "users with admin privileges" could be clearer in the context of Teleport, since this page uses them for the first time here. Maybe add links around these terms to more information about token UUIDs and RBAC? On a similar note, in the sentence,"Don't give users the `admin` role when giving them the more restrictive `access,editor` roles will do instead," we could add links to more information about these roles.~

  - ~"Saving tokens into a file rather than sharing tokens directly as strings": I was not sure if this was a recommendation for or against doing this. I was also not sure what tokens were in this context, and whether "directly as strings" meant "in memory" or using some other kind of inter-process communication.~

  - ~While the first tip box includes a link to a page on RBAC best practices ("Learn more about Teleport Role-Based Access Control best practices."), the page turned out to be a general index of pages related to access control, rather than the guide to best practices I was expecting.~

- When entering the `helm install` command in the first snippet, I received the error, `CLUSTER_NAME: parameter not set` because of the use of the `?` in the command’s parameter expansion. I also noticed an `EMAIL` environment variable that also needed to be populated. 

- In the "Install Teleport" section, the `jq` command for creating a changeset in AWS spans multiple lines, meaning that the single-line copy button within the code snippet only copies the first line.

### Category
- Improve Existing

### LINKED ISSUE #9112: Make the "Prerequisites" section clearer in the "Getting Started - Kubernetes with SSO" guide
## Details
I ran into some issues while following the ["Getting Started - Kubernetes with SSO"](https://goteleport.com/docs/kubernetes-access/getting-started/cluster/) guide. The issues had to do with preparing my environment, and it would be great if the ["Prerequisites" section](https://goteleport.com/docs/kubernetes-access/getting-started/cluster/) were more specific about the kind of setup required here. 

The Prerequisites section indicates that the guide requires only Kubernetes and Helm. For speed and ease of access, I decided to use a [kind](https://kind.sigs.k8s.io/) cluster on my local machine, i.e., without a registered domain name or cloud provider. However, this presented some issues.

- The first `helm install` command enables TLS via Let's Encrypt and the ACME protocol, but I didn't have a registered domain name or an ingress to my Teleport instance.

- The command `MYIP=$(kubectl get services teleport-cluster -o jsonpath='{.status.loadBalancer.ingress[0].ip}')` returned an empty result. The quickstart guide included a link to the Kubernetes documentation on external load balancers. After reading the page, I realized that only cloud providers that have implemented the external load balancer functionality can assign an external IP based on a Service manifest with `type: LoadBalancer`. According to the kind documentation, the best workaround on my Mac would be to [use an Ingress](https://kind.sigs.k8s.io/docs/user/loadbalancer/) instead of an external load balancer. However, the `teleport-cluster` Helm chart [hardcodes](https://github.com/gravitational/teleport/blob/a28bb903fb2caf4b54acd48bebf67d0d987b2741/examples/chart/teleport-cluster/templates/service.yaml#L20) the type: LoadBalancer` field in its Service manifest, and I did not want to explore alternatives that would mutate the default Helm chart.

I tried the guide again after spinning up a bare-bones EKS cluster via [eksctl](https://eksctl.io/). Here are some issues I ran into:

- After running the `helm install` command successfully in my EKS cluster, the command `MYIP=$(kubectl get services teleport-cluster -o jsonpath='{.status.loadBalancer.ingress[0].ip}')` had an empty output. This was because, in my EKS cluster, the `hostname` field was populated rather than the `ip` field.

- I received the following error when executing the `aws route53 change-resource-record-sets` command with **myrecords.json** (created via the "AWS Route 53" tab [here](https://goteleport.com/docs/kubernetes-access/getting-started/cluster/#step-13-install-teleport)) and my EKS cluster:
  
  ```
  An error occurred (InvalidChangeBatch) when calling the ChangeResourceRecordSets operation: [Invalid Resource Record: 'FATAL problem: ARRDATAIllegalIPv4Address (Value is not a valid IPv4 address) encountered with '<redacted>'', RRSet with DNS name tele.example.com. is not permitted in zone <redacted>., Invalid Resource Record: 'FATAL problem: ARRDATAIllegalIPv4Address (Value is not a valid IPv4 address) encountered with '<redacted>'', RRSet with DNS name \052.tele.example.com. is not permitted in zone <redacted>.]
  ```

- I created a user and attempted to access the URL returned by the `tctl users add` command. Because I had not registered a domain name for the Teleport server or succeeded in creating DNS records via Route 53, I replaced the hostname with the external IP value of the teleport-cluster service (keeping the `https://` scheme and path). However, when I sent an HTTP request to this URL, the response was a 302 with a `Location` header that changed the hostname to `teleport-cluster`, causing my browser to return an NXDOMAIN error. I was able to exec into the teleport-cluster pod and run `curl --insecure https://localhost:3080/web/invite/<token redacted>` (i.e., with the invitation URL) and got the expected HTML body, though without a browser, I could go no further from there.

### Category
- Improve Existing

### PR DESCRIPTION
**Pull request by ptgott**
This change address some issues I found while working through
the "Getting Started - Kubernetes with SSO" guide.

- Assign required env variables in code snippets, and offer
  instructions for changing them. Otherwise, a reader may not
  realize that these variables are required for a command when
  pasting code.

- Put the AWS Route 53 jq command on one line. This way, users
  can copy it using our single-line copy button, and most of
  the command will no longer display as a comment.

- Add an admonition that describes the difference between this
  guide and our other Kubernetes Getting Started guide. This
  way, a reader won't have to read through that guide to
  understand why we have two Getting Started guides for
  Kubernetes.

- Indicate that a domain name is required. Not having one
  really tripped me up when I worked through this guide the
  first time.

- For users with environments that preclude access to the public
  internet, I added an admonition linking to the docker-compose
  guide and our contact page, with a mention of our on-prem
  Kubernetes support. We can replace this with a link to a
  local-only Kubernetes guide when one is available.

- While some Kubernetes load balancers populate the hostname,
  rather than the IP in the status field, this guide only
  includes instructions for fetching the IP. This adds
  instructions for fetching the hostname as well.

  See:
  https://kubernetes.io/docs/reference/kubernetes-api/service-resources/service-v1/#ServiceStatus

Closes #9086
Closes #9112

### CODE REVIEW

File: docs/pages/kubernetes-access/getting-started/cluster.mdx [hunk 1]
```diff
@@ -5,12 +5,18 @@ description: Getting started with Teleport. Let's deploy Teleport in a Kubernete
 
 # Getting Started
 
-Let's deploy Teleport in a Kubernetes with SSO and Audit logs:
+Let's deploy Teleport in a Kubernetes cluster with SSO and Audit logs:
```
Comment (zmb3) [inline docs/pages/kubernetes-access/getting-started/cluster.mdx:8]: Readers might get a little bit confused that we're using the same kubernetes cluster as both:

- a place to deploy teleport
- the resource that teleport will provide access to

Can we try to clarify this a bit? Maybe saying something like:

"Teleport can provide secure, unified access to your Kubernetes clusters. This guide assumes you don't have an existing Teleport cluster running, and will deploy Teleport on the same Kubernetes cluster you'll provide access to."

File: docs/pages/kubernetes-access/getting-started/cluster.mdx [hunk 2]
```diff
@@ -23,18 +29,29 @@ Let's deploy Teleport in a Kubernetes with SSO and Audit logs:
...
 
 ## Step 1/3. Install Teleport
 
-Let's start with a single-pod Teleport using persistent volume as a backend.
+Let's start with a single-pod Teleport using persistent volume as a backend. Modify the values of `CLUSTER_NAME` and `EMAIL` according to your environment, where `CLUSTER_NAME` is the domain name you are using for your Teleport deployment and `EMAIL` is an email address used for notifications.
 
 <Tabs>
   <TabItem label="Open Source">
     ```code
+    $ CLUSTER_NAME="tele.example.com"
```
Comment (zmb3) [inline docs/pages/kubernetes-access/getting-started/cluster.mdx:53]:
```suggestion
    $ export CLUSTER_NAME="tele.example.com"
```

Export is required in order for `CLUSTER_NAME` to remain set for future commands.

Comment (ptgott) [reply docs/pages/kubernetes-access/getting-started/cluster.mdx:53]: Since `CLUSTER_NAME` and `EMAIL` are only used in the initial shell environments of the two code snippets, and their values are passed to the `helm install` command via parameter expansion (i.e. the variables themselves aren't part of the environment of `helm install`), would it be safer to leave these variables unexported?

**Review by zmb3 — APPROVED** [review]

File: docs/pages/kubernetes-access/getting-started/cluster.mdx [hunk 3]
```diff
@@ -23,18 +28,29 @@ Let's deploy Teleport in a Kubernetes with SSO and Audit logs:
...
 
 ## Prerequisites
+- A registered domain name. This is required for Teleport to set up TLS via Let's Encrypt and for Teleport clients to verify the Proxy Service host.
+
+- A Kubernetes cluster hosted by a cloud provider, which is required for the load balancer we deploy in this guide.
+
+<Admonition title="Local-only setups" type="tip">
+Teleport also supports Kubernetes in on-premise and air-gapped environments. If you would like to try out Teleport on your local machine, we recommend following our [Docker Compose guide](../../getting-started/docker-compose.mdx). 
+
+You can also [contact us](/contact-us/) so we can work with you to determine the best setup for your Kubernetes environment.
```
Comment (r0mant) [inline docs/pages/kubernetes-access/getting-started/cluster.mdx:38]: I'd remove this sentence TBH. Seems irrelevant to the guide.

File: docs/pages/kubernetes-access/getting-started/cluster.mdx [hunk 4]
```diff
@@ -101,6 +125,12 @@ to create a public IP for Teleport.
     $ echo $MYIP
     # 192.168.2.1
     ```
+
+    If `$MYIP` is blank, your cloud provider may have assigned a hostname to the load balancer rather than an IP address. Run the following command to retrieve the hostname:
+
+    ```code 
+    $ MYHOSTNAME=$(kubectl get services teleport-cluster -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
```
Comment (r0mant) [inline docs/pages/kubernetes-access/getting-started/cluster.mdx:132]: How is `MYHOSTNAME` used? It doesn't seem to be used below, only `MYIP` is.

**Review by r0mant — APPROVED** [review]
